### ISSUE TITLE
cc on empty line does not indent properly 

### ISSUE THREAD

**Issue by jasonify** [VSCodeVim/Vim#800]
It just doesn't work.

**Comment by sectioneight** [comment]
@jasonify can you follow the issue submission guideline? It's hard to help when all we have is "It just doesn't work". It's working just fine for me.

Your report should look like this:

### What did you do?

Put the exact keys you pressed.

### What did you expect to happen?

### What happened instead?

### Technical details:

* VSCode Version:
* VsCodeVim Version: *[please ensure you are on the latest]*
* OS:

**Comment by johnfn** [comment]
@sectioneight, @jasonify is actually a good friend of mine and was just trolling me 😀 

The problem he found is that if you do `cc` on an empty line it should indent to the correct level by default. I'm pretty sure it happens on all VSC's and all Vim versions.

**Comment by sectioneight** [comment]
lol sorry

**Comment by johnfn** [comment]
Haha, it's fine, your response totally cracked me up. 😀

**Comment by johnfn** [comment]
I just tried to do this and it's actually much harder than I thought. VSCode does not provide any API I can find for directly indenting a line. 

* I tried replacing it with `ddO` (because `editor.action.insertLineBefore` indents correctly) but that's actually not quite the same in all cases. What if the next line was also an empty line and had no spaces at all?
* Then I tried replacing it with something like `C<esc>ix<esc>==x` (clear the line, insert an x and then format the x to the right indentation, then delete the x. I considered this for about 5 seconds before concluding it was an enormous hack.

I'm actually kind of stuck on this one unless someone can find a better VSCode command API I could use.
